File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
import sys

input = sys.stdin.readline
write = sys.stdout.write

def sq(depth: int, start: int, m: int, result: list[str], ans: list[int], n: int):
    if depth == m:
        result.append(" ".join(map(str, ans)))
        return
    
    for i in range(start, n + 1):
        ans.append(i)
        
        sq(depth + 1, i, m, result, ans, n)
            
        ans.pop()

            
def solve():
    line = input().split()
    if not line: return
    n, m = map(int, line)
    
    result = []
    ans = []
    
    sq(0, 1, m, result, ans, n)
    
    write("\n".join(result) + "\n")

if __name__ == "__main__":
    solve()

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.cc
#include <stdio.h>
#include <stdlib.h>

int ans[10];

void sq(int depth, int start, int m, int n) {
	if (depth == m) {
		for (int i = 0; i < m; i++) {
			printf((i == m - 1 ? "%d\n" : "%d "), ans[i]);
		}
		return;
	}

	for (int i = start; i <= n; i++) {
		ans[depth] = i;

		sq(depth + 1, i, m, n);
	}
}

int main(void) {
	int n = 0, m = 0;
	if (scanf("%d %d", &n, &m) != 2) return 0;

	sq(0, 1, m, n);

	return 0;
}